android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt

```diff
@@ -420,31 +420,6 @@ constructor(
     completed * 100 / if (total > 0) total else 1
 
   suspend fun schedulePeriodicJobs(context: Context) {
-    if (!BuildConfig.SKIP_AUTHENTICATION) {
-      if (context.isDeviceOnline()) {
-        // Do not schedule sync until location selected when strategy is RelatedEntityLocation
-        // Use applicationConfiguration.usePractitionerAssignedLocationOnSync to identify
-        // if we need to trigger sync based on assigned locations or not
-        when {
-          applicationConfiguration.syncStrategy.contains(SyncStrategy.RelatedEntityLocation) -> {
-            if (
-              applicationConfiguration.usePractitionerAssignedLocationOnSync ||
-                context
-                  .retrieveRelatedEntitySyncLocationState(MultiSelectViewAction.SYNC_DATA)
-                  .isNotEmpty()
-            ) {
-              syncBroadcaster.schedulePeriodicSync(applicationConfiguration.syncInterval)
-            }
-          }
-          else -> syncBroadcaster.schedulePeriodicSync(applicationConfiguration.syncInterval)
```

Comment:
using the original code order.
try to wrap schedulePeriodicSync with viewModelScope.launch.

the original code might be hold down on this line, since schedulePeriodicSync is a suspend function and the process inside might block the next process